### PR TITLE
Chore/SOF-6774-ts

### PR DESCRIPTION
**Pull request by spectre007**
This PR concerns the addition of the TypeScript layer to the `standata` repository

### CODE REVIEW

File: src/js/application.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Standata } from "./base";
+import APPLICATIONS from "./entities/applications.json";
```
Comment (timurbazhirov) [inline src/js/application.ts:2]: entities -> runtime_data

File: src/js/base.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    [key: string]: string[];
+}
+
+export interface StandataConfig {
+    categories: EntityCategories;
+    entities: EntityItem[];
+}
+
+export class Standata {
+    entities: EntityItem[];
```
Comment (timurbazhirov) [inline src/js/base.ts:16]: static runtimeData = {};

File: src/js/application.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export class ApplicationStandata extends Standata {
+    constructor() {
```
Comment (timurbazhirov) [inline src/js/application.ts:5]: export class ApplicationStandata extends Standata {
    static runtimeData = APPLICATIONS;
}

File: src/js/base.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                } else {
+                    lookupTable[category] = new Set([filename]);
+                }
+            }
+        }
+        return lookupTable;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    protected loadEntity(filename: string): object | undefined {
```
Comment (timurbazhirov) [inline src/js/base.ts:63]:
```js
    protected loadEntity(filename: string): object | undefined {
        return this.constructor.runtimeData[filename];
    }
```

File: README.md [hunk 5]
```diff
@@ -44,4 +44,27 @@ standata-symlinks --help
...
+
+# creates symbolic links in materials/by_category (ts-node)
+ts-node src/js/cli.ts materials/categories.yml
+
+# creates symbolic links for materials in tmp/by_category
+ts-node src/js/cli.ts -d tmp materials/categories.yml
+
+# run via npm
+npm run build:categories -- materials/categories.yml
+
```
Comment (timurbazhirov) [inline README.md:69]: Let's explain how to generate the `json` funtime data files pls

File: src/js/base.ts (hunk 4 above)
Comment (spectre007) [reply src/js/base.ts:73]: A workaround is needed here for TypeScript (https://stackoverflow.com/questions/33387318/access-to-static-properties-via-this-constructor-in-typescript)
```typescript
this.constructor as typeof Standata
```

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**